tests/Tests/QuickCheckUtils.hs

```diff
@@ -79,6 +80,10 @@ instance Arbitrary B.ByteString where
     arbitrary     = B.pack `fmap` arbitrary
     shrink        = map B.pack . shrink . B.unpack
 
+instance Arbitrary BL.ByteString where
+    arbitrary = BL.fromChunks <$> arbitrary
```

Comment:
```suggestion
    arbitrary = oneof
      [ BL.fromChunks <$> arbitrary
      -- so that a single utf8 code point could appear split over up to 4 chunks
      , BL.fromChunks . map B.singleton <$> arbitrary
      -- so that a code point with 4 byte long utf8 representation
      -- could appear split over 3 non-singleton chunks
      , (\a b c -> BL.fromChunks [a, b, c])
        <$> arbitrary
        <*> ((\a b -> B.pack [a, b]) <$> arbitrary <*> arbitrary)
        <*> arbitrary
      ]
```
So that we test even most malicious strings.

Reply:
Good idea. I also tweaked the generators to not use `arbitrary` for `Word8`, which is size-based (which makes short strings ASCII only), opting instead for a simple uniform distribution.